jwst/tweakreg/astrometric_utils.py

```diff
@@ -81,8 +86,15 @@ def create_astrometric_catalog(input_models, catalog="GAIADR3", output="ref_cat.
     radius, fiducial = compute_radius(outwcs)
 
     # perform query for this field-of-view
-    ref_dict = get_catalog(fiducial[0], fiducial[1], sr=radius, catalog=catalog)
-    colnames = ('ra', 'dec', 'mag', 'objID')
+    epoch = (
+        epoch
+        if epoch is not None
+        else input_models[0].meta.target["proper_motion_epoch"]
```

Comment:
OK, sometimes I'm a little thick, but I'm not sure I understand why you would want to use the epoch that was the reference for the target coords given in the proposal, rather than using the actual time/epoch of the observation itself, e.g. meta.observation.date_beg (which unfortunately is a date-time string, so you'd have to parse it to convert to floating-point year).

Reply:
I wondered the same thing but decided to follow the logic used by the roman folks. @mairanteodoro may have a better explanation for this choice.

Reply:
Sorry, I put too much faith in the similarity of the codebases and I got lazy in my testing; won't happen again. Roman gathers the epoch metadata in a different way than JWST. I'll add code to get the correct metadata.